wgsl/index.bs

```diff
@@ -5339,6 +5339,101 @@ Issue: [[#uniform-control-flow]] needs to state whether all invocations being di
   </xmp>
 </div>
 
+### Control Flow Interruption ### {#control-flow-interruption-section}
+
+A <dfn>control flow interruption</dfn> is any one of the following statements:
+* a [=break=] statement
+* a [=continue=] statement
+* a [=fallthrough=] statement
+* a [=return=] statement
+* a [=discard=] statement
+* a [=compound statement=] containing a [=control flow interruption=].
+
+A control flow interruption must appear last in its immediately enclosing compound statement.
+
+Note: Statements that immediately follow a control flow interruption,
+but within the same enclosing compound statement, can never be executed.
+These may indicate a programming error.
+
+[=Switch=], [=if=], [=loop=], and [=for=] statements *contain* [=compound statements=],
```

Comment:
so, just to clarify, the SPIR-V code generated for this construct will be fine?
```rust
switch(0) {
  default: return;
}
foo = 5;
```

IIRC, you mentioned that if we generate `OpSelectionMerge` block faithfully, and not try to optimize it, the SPIR-V module would still be correct.

Maybe this could be one of the examples?

Reply:
Yes, the equivalent code is valid in SPIR-V and in GLSL.

I just tried it with this GLSL fragment shader:
```glsl
#version 450

void main() {
  int foo;
  switch (0) {
   default: return;
  }
  foo = 5;
}
```

Something to watch out for though:  Glslang sees that the 'foo=5' is trivially unreachable (statically, without checking condition values), and never emits it.  So if you're trying to generate the SPIR-V case, you have to go out of your way to add  a non-trivial instruction.
(And once I saw that, I remember that *I'm* the one who changed Glslang about a year ago to skip those statically unreachable blocks.)

It's a good example, and I'll add it.